sympy/simplify/simplify.py

```diff
@@ -1758,7 +1758,7 @@ def compare(s, alt_s):
         # get the non-commutative part
         c_args, args = expr.args_cnc()
         com_coeff = Mul(*c_args)
-        if com_coeff != 1:
+        if com_coeff != 1 and com_coeff != 1.0:
```

Comment:
We might get floats of different precision.

It would be better to use `from sympy.core.numbers import equal_valued`.